sdk/src/main/java/com/karhoo/sdk/api/service/loyalty/LoyaltyBurnPointsInteractor.kt

```diff
@@ -0,0 +1,33 @@
+package com.karhoo.sdk.api.service.loyalty
+
+import com.karhoo.sdk.api.KarhooError
+import com.karhoo.sdk.api.datastore.credentials.CredentialsManager
+import com.karhoo.sdk.api.model.LoyaltyPoints
+import com.karhoo.sdk.api.network.client.APITemplate
+import com.karhoo.sdk.api.network.response.Resource
+import com.karhoo.sdk.api.service.common.BaseCallInteractor
+import kotlinx.coroutines.CompletableDeferred
+import kotlinx.coroutines.Deferred
+import kotlinx.coroutines.Dispatchers
+import javax.inject.Inject
+import kotlin.coroutines.CoroutineContext
+
+internal class LoyaltyBurnPointsInteractor @Inject constructor(credentialsManager: CredentialsManager,
+                                                           private val apiTemplate: APITemplate,
+                                                           context: CoroutineContext = Dispatchers.Main
+                                                          ) :
+    BaseCallInteractor<LoyaltyPoints>(true, credentialsManager, apiTemplate, context) {
+
+    internal var loyaltyId: String? = null
```

Comment:
the properties can be made private, don't think there's a reason to be made internal.

Reply:
They can be passed in from the constructor

Reply:
Same applies for the below properties in the LoyaltyPointsToEarnInteractor